### DISCUSSION TITLE
How to use BLE? (EDIT - repo partially works with JK-B2A8S20P board via BLE)

### DISCUSSION THREAD

**Discussion by EasilyBoredEngineer** [syssi/esphome-jk-bms#110] [Q&A] [answered]
Hi @syssi -

I've received my ESP32, installed home assistant then stumbled my way around around the YAML and seem to have your component installed on the ESP32.

What I'm wondering though... how to configure the yaml to use BLE JK02?

Thankyou!

EBE

**Comment by EasilyBoredEngineer** [comment, marked as answer]
Please disregard this @syssi - I've stumbled some more and think I'm on the right track...

For others, the appropriate BLE example yaml looks to be at [the following code example](https://github.com/syssi/esphome-jk-bms/blob/main/esp32-ble-example.yaml)

**Reply by EasilyBoredEngineer** [reply]
@syssi as an fyi this 'sorta works' with the JK-B2A8S20P bms and BLE.

Attached please find log file and some screenshots. Many of the bizarre flags are likely owing to the fact that this is simply on the bench, on a 4S bank, and hasn't yet completed a charging cycle.

[logs_testesp_logs.txt](https://github.com/syssi/esphome-jk-bms/files/8692327/logs_testesp_logs.txt)
![Screenshot 2022-05-14 153150](https://user-images.githubusercontent.com/105184462/168412339-74cd17f3-aaa4-49a9-992c-b3af71d070e1.jpg)
![Screenshot 2022-05-14 153121](https://user-images.githubusercontent.com/105184462/168412342-f590d490-29dc-4aca-952f-8874f79f0ee9.jpg)
![Screenshot 2022-05-14 153041](https://user-images.githubusercontent.com/105184462/168412344-d8ab3d0c-c3bd-441e-9e46-50565efc6b08.jpg)
![Screenshot 2022-05-14 152946](https://user-images.githubusercontent.com/105184462/168412346-6db012f1-f17b-4e17-bda4-02f77a865666.jpg)


linking back [to the other discussion](https://github.com/jblance/mpp-solar/issues/224) for completeness.

**Reply by syssi** [reply]
I've created an issue for your device model a few days ago here: https://github.com/syssi/esphome-jk-bms/issues/105

If you find something odd please add it to the issue. For example: You could help to decode the different bits of the error bitmask above by providing a screenshot of the Android app showing the same errors. We could fix the errors step by step (change to battery string to 4S, adjust the undertemperature protection to a new value and so on). I hope we can identify most of the errors reported above.

**Reply by syssi** [reply]
I assume one error bit indicates something like: "Please set the nominal capacity of your battery bank at the settings". If you do so one of the "unknown errors" above should disappear.

**Comment by EasilyBoredEngineer** [comment]


**Reply by EasilyBoredEngineer** [reply]
(PS - I've checked out your code base... I wish I even had a pinch of your coding skills - impressed! very neat)

**Reply by syssi** [reply]
Thanks for the flowers! :-)
